.github/workflows/ci.yml

```diff
@@ -56,6 +56,14 @@ jobs:
       - name: Checkout
         uses: actions/checkout@master
 
+      - name: "Cache dependencies installed with composer"
+        uses: actions/cache@v1.0.0
+        with:
+          path: ~/.composer/cache
+          key: ${{ matrix.php-binary }}-composer-${{ matrix.dependencies }}-${{ hashFiles('**/composer.json') }}
```

Comment:
the problem with `${{ hashFiles('**/composer.json') }}` here is it does not allow the cache to be additive when `matrix.dependencies == 'highest'` starts downloading **new** dependencies.
GitHub Actions will not overwrite the cache so some deps will never be cached.
The cache key needs to be unique to fix this.
Using this would allow the cache to be additive over time:
`${{ matrix.php-binary }}-composer-${{ matrix.dependencies }}-${{ github.sha }}`

Reply:
They will be used, as the `restore-keys` keys will be used to find the latest matching cache hit by substring.
They won't be used for long, but that doesn't matter, more recent ones will be.